CONTRIBUTING.md

```diff
@@ -56,6 +56,25 @@ Button.propTypes = {
 };
 ```
 
+### TypeScript
+
+### Type definitions for Props and States
+Always use `interface` for type definitions Props and States. Interfaces are more “extendable” due to the support of declaration merging. It is useful when components supports native HTML props, for example:
+
+```javascript
+interface Props extends <HTMLInputElement> {
+    /** Description of this props  */
+    context: string;
+}
+```
+
+### Using Type
+Types are useful for union types. So its are more “composable” :
```

Comment:
Little grammar check 2

Types are useful for union types, meaning they are more "composable":

Reply:
Done